src/ompl/geometric/planners/bitstar/datastructures/src/IntegratedQueue.cpp

```diff
@@ -0,0 +1,1693 @@
+                this->rmOutgoingLookup(oldEdgeIter);
+            }
+            //No else
+
+            //Finally erase from the queue:
+            edgeQueue_.erase(oldEdgeIter);
+        }
+
+
+
+        void BITstar::IntegratedQueue::rmIncomingLookup(const edge_queue_iter_t& mmapIterToRm)
+        {
+            if (incomingLookupTables_ == true)
+            {
+                this->rmEdgeLookupHelper(incomingEdges_, mmapIterToRm->second.second->getId(), mmapIterToRm);
+            }
+            //No else
+        }
+
+
+
+        void BITstar::IntegratedQueue::rmOutgoingLookup(const edge_queue_iter_t& mmapIterToRm)
+        {
+            if (outgoingLookupTables_ == true)
+            {
+                this->rmEdgeLookupHelper(outgoingEdges_, mmapIterToRm->second.first->getId(), mmapIterToRm);
+            }
+            //No else
+        }
+
+
+
+        void BITstar::IntegratedQueue::rmEdgeLookupHelper(vid_edge_queue_iter_umap_t& lookup, const BITstar::vid_t& idx, const edge_queue_iter_t& mmapIterToRm)
+        {
+            //Variable:
+            //An iterator to the vertex,list pair in the lookup
+            vid_edge_queue_iter_umap_t::iterator iterToVertexListPair;
+
+            //Get the list in the lookup for the given index:
+            iterToVertexListPair = lookup.find(idx);
+
+            //Make sure it was actually found before derefencing it:
+            if (iterToVertexListPair != lookup.end())
+            {
+                //Variable:
+                //Whether I've found the mmapIterToRm in my list:
+                bool found;
+                //The iterator to the mmapIterToRm in my list:
+                edge_queue_iter_list_t::iterator iterToList;
+
+                //Start at the front:
+                iterToList = iterToVertexListPair->second.begin();
+
+                //Iterate through the list and find mmapIterToRm
+                found = false;
+                while(found == false && iterToList != iterToVertexListPair->second.end())
+                {
+                    //Compare the value in the list to the target:
+                    if (*iterToList == mmapIterToRm)
+                    {
+                        //Mark as found:
+                        found = true;
+                    }
+                    else
+                    {
+                        //Increment the iterator:
+                        ++iterToList;
+                    }
+                }
+
+                if (found == true)
+                {
+                    iterToVertexListPair->second.erase(iterToList);
+                }
+                else
+                {
+                    throw ompl::Exception("Edge iterator not found under given index in lookup hash.");
+                }
+            }
+            else
+            {
+                throw ompl::Exception("Indexing vertex not found in lookup hash.");
+            }
+        }
+
+
+
+
+
+
+
+
+
+
```

Comment:
I see your point, but I do not fully subscribe to the rules your propose. 1 space for functions, if you have different parts, makes sense sometimes, but I would expect that together with a comment maybe?
I think 1 line between functions is ample, since you get an extra almost empty line with "}" above. That said, I do not mind at all adding more space, so I can be convinced your rule is better. However, for consistency with the rest of OMPL it might be easier to follow the same style. I am not asking for this just to be particular, but if we follow a consistent style it is easier for others to read & understand the code.
Lastly, a bunch of space between groups of functions is not sufficiently well defined. Maybe add one empty line with a comment with a an additional commented marker make of - or = or / ? Too much empty space will make think the file is done :)